hoot-core/src/main/cpp/hoot/core/conflate/polygon/MultiPolygonCreator.cpp

```diff
@@ -221,6 +229,171 @@ boost::shared_ptr<Geometry> MultiPolygonCreator::createMultipolygon() const
   return result;
 }
 
+QString MultiPolygonCreator::_findRelationship(LinearRing *ring1, LinearRing *ring2) const
+{
+  QString result = "";
+
+  const GeometryFactory& gf = *GeometryFactory::getDefaultInstance();
+  vector<Geometry*> noHoles;
+  boost::shared_ptr<Polygon> p1(gf.createPolygon(*ring1, noHoles));
+  boost::shared_ptr<Polygon> p2(gf.createPolygon(*ring2, noHoles));
+
+  // It would be nice to do this in one call but "isWithin" doesn't seem to return anything.
+  // This could be refactored down to isContains, else, make p2, isContains.
+
+  boost::shared_ptr<IntersectionMatrix> im(p1->relate(ring2));
+  if (im->isContains())
+  {
+    result = "outer";
+  }
+//  else if (im->isWithin())
+//  {
+//    result = "inner";
+//  }
+
+  boost::shared_ptr<IntersectionMatrix> im2(p2->relate(ring1));
+  if (im2->isContains())
+  {
+    result = "inner";
+  }
+//  else if (im2->isWithin())
+//  {
+//    result = "outer";
+//  }
+
+  return result;
+}
+
+void MultiPolygonCreator::_classifyRings(std::vector<LinearRing *> &noRole,
+                                         std::vector<LinearRing *> &inners,
+                                         std::vector<LinearRing *> &outers) const
+{
+  // Empty == nothing else to do
+  if (noRole.size() == 0)
+  {
+    return;
+  }
+
+  // One polygon, no inners or outers
+  if (noRole.size() == 1 && inners.size() == 0 && outers.size() == 0)
+  {
+    outers.push_back(noRole[0]);
+    return;
+  }
+
+  // A list of things we haven't found yet.
+  deque<LinearRing *> notFound;
+
+  for (size_t i = 0; i < noRole.size(); ++i)
+  {
+    QString status = "";
+
+    // Check of we have an inner polygon.
+    for (size_t j = 0; j < outers.size(); ++j)
+    {
+      status = _findRelationship(noRole[i],outers[j]);
+
+      if (status == "inner")
+      {
+        inners.push_back(noRole[i]);
+        break;
+      }
+      else if (status == "outer")
+      {
+        // This is not a good combination: Outside of an Outer
+        outers.push_back(noRole[i]);
+        break;
+      }
+    }
+
+    // No outers or we didn't find an inner polygon
+    if (status == "")
+    {
+      // Look for an outer polygon
+      for (size_t j = 0; j < inners.size(); ++j)
+      {
+        status =_findRelationship(noRole[i],inners[j]);
+
+        if (status == "inner")
+        {
+          // This is not a good combination: Inside of an inner
+          inners.push_back(noRole[i]);
+          break;
+        }
+        else if (status == "outer")
+        {
+          outers.push_back(noRole[i]);
+          break;
+        }
+      }
+
+      // No inners or we still didn't find anything so push it onto the list of things to check later
+      if (status == "")
+      {
+        notFound.push_back(noRole[i]);
+      }
+    }
+  }
+
+  // Now go through the things we didn't find.
+
+  // Early return?
```

Comment:
I think this check is unnecessary, as the rest of the code in this function is in the following while loop - which will not execute if `notFound.size() == 0`

Reply:
Good catch. Removing it.